File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int countOne = getCountOne(n);
        int count = 0;
        while(countOne != count){
            n++;
            count = getCountOne(n);
        }
        return n;
    }
    
    private int getCountOne(int num){
        return Integer.toBinaryString(num).replace("0", "").length();
    }
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    answer = 0
    cnt = getBin(n)
    while True :
        n += 1
        if cnt == getBin(n) :
            answer = n
            break;
    return answer

def getBin(num) :
    return str(bin(num)[2:]).count("1")